corehq/apps/groups/views.py

```diff
@@ -160,6 +171,16 @@ def _update_group_membership(request, domain, group_id):
     safe_ids = [u for u in selected_users if u in all_users]
 
     group.users = safe_ids
+
+    _ensure_case_sharing_privilege(request, group)
+
     group.save()
     messages.success(request, _("Group %s updated!") % group.name)
     return HttpResponseRedirect(reverse("group_members", args=[domain, group_id]))
+
+
+def _ensure_case_sharing_privilege(request, group):
```

Comment:
This approach, where groups get disabled as they're individually saved, seems confusing. Did y'all decide it was a bad idea to deliberately switch all groups to case sharing on downgrade?

Reply:
yes we wanted to keep case sharing groups as is during a downgrade and not switch them all to reporting groups as that could have a negative affect on existing mobile workers / CHWs using the app.